File: simple_user_test.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import requests
import json
import time

BASE_URL = "https://localhost:7157"

def test_user_api():
    """简单测试用户API"""
    print("测试修复后的用户API")
    
    login_data = {
        "username": "sysadmin",
        "password": "Admin@123456",
        "rememberMe": True
    }
    
    session = requests.Session()
    session.verify = False  # 忽略SSL证书验证
    try:
        # 登录
        print("正在登录...")
        response = session.post(
            f"{BASE_URL}/api/v1.0/Auth/login",
            json=login_data,
            headers={"Content-Type": "application/json"},
            timeout=10
        )
        
        if response.status_code != 200:
            print(f"登录失败: {response.status_code}")
            return
            
        result = response.json()
        if not result.get('success'):
            print(f"登录失败: {result.get('message', 'N/A')}")
            return
            
        token = result['data']['token']
        print("登录成功")
        
        headers = {
            "Content-Type": "application/json",
            "Authorization": f"Bearer {token}"
        }
        
        # 测试Users API
        print("测试Users API...")
        
        user_response = session.get(
            f"{BASE_URL}/api/v1.0/Users",
            headers=headers,
            timeout=10
        )
        
        print(f"状态码: {user_response.status_code}")
        
        if user_response.status_code == 500:
            result = user_response.json()
            error_msg = result.get('message', '')
            print(f"500错误: {error_msg}")
            
            if '列名' in error_msg and '无效' in error_msg:
                print("字段映射问题仍然存在")
                return False
            else:
                print("字段映射问题已解决，但存在其他错误")
                return False
                
        elif user_response.status_code == 200:
            print("SUCCESS: Users API修复成功!")
            result = user_response.json()
            if result.get('success'):
                data = result.get('data', {})
                if isinstance(data, dict) and 'items' in data:
                    users = data['items']
                    print(f"返回 {len(users)} 条用户记录")
                    
                    if users:
                        first_user = users[0]
                        print("用户字段:")
                        print(f"  username: {first_user.get('username', 'N/A')}")
                        print(f"  realName: {first_user.get('realName', 'N/A')}")
                        print(f"  department: {first_user.get('department', 'N/A')}")
                        print(f"  position: {first_user.get('position', 'N/A')}")
                        print(f"  remark: {first_user.get('remark', 'N/A')}")
                        print(f"  updateTime: {first_user.get('updateTime', 'N/A')}")
                    return True
        else:
            print(f"未知状态码: {user_response.status_code}")
            return False
            
    except requests.exceptions.ConnectionError:
        print("无法连接到服务器")
        return False
    except Exception as e:
        print(f"测试失败: {str(e)}")
        return False

if __name__ == "__main__":
    time.sleep(2)
    success = test_user_api()
    if success:
        print("\n=== 修复验证成功 ===")
    else:
        print("\n=== 修复验证失败 ===")

File: simple_user_test.js
const https = require('https');
process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';

async function simpleUserUpdateTest() {
  try {
    console.log('简单用户更新测试...\n');

    // 1. 获取Token
    const token = await getAuthToken();
    console.log('✅ 获取Token成功\n');

    // 2. 创建新用户
    console.log('📝 创建测试用户...');
    const newUserData = {
      Username: 'testuser_' + Date.now(),
      RealName: '测试用户',
      Email: 'test@example.com',
      Phone: '13900000000',
      Role: 'Doctor',
      IsActive: true,
      Remark: '测试创建'
    };

    const createResponse = await apiRequest('POST', '/api/v1/users', token, newUserData);
    console.log('创建用户响应:', createResponse);
    
    if (!createResponse.success) {
      console.log('❌ 创建用户失败:', createResponse.error);
      return;
    }

    const createdUserId = createResponse.data && createResponse.data.data ? createResponse.data.data.id : createResponse.data.id;
    console.log('✅ 用户创建成功，ID:', createdUserId);

    // 3. 立即查询验证创建
    const getUserResponse = await apiRequest('GET', `/api/v1/users/${createdUserId}`, token);
    if (getUserResponse.success) {
      const user = getUserResponse.data.data;
      console.log('📊 创建后的用户信息:');
      console.log(`   用户名: ${user.username}`);
      console.log(`   真实姓名: ${user.realName}`);
      console.log(`   邮箱: ${user.email}`);
      console.log(`   手机: ${user.phoneNumber}`);
      console.log(`   角色: ${user.role}`);
      console.log(`   备注: ${user.remark}`);
    }

    // 4. 更新用户数据
    console.log('\n🔄 测试更新用户数据...');
    const updateData = {
      Id: createdUserId,
      Username: newUserData.Username,
      RealName: '测试用户_已更新',
      Email: 'updated@example.com',
      Phone: '13800000000',
      Role: 'Doctor',
      IsActive: true,
      Remark: '测试更新完成'
    };

    const updateResponse = await apiRequest('PUT', `/api/v1/users/${createdUserId}`, token, updateData);
    console.log('更新响应状态:', updateResponse.statusCode);
    console.log('更新响应内容:', updateResponse.data);

    if (updateResponse.success) {
      console.log('✅ 更新请求成功');
      
      // 5. 再次查询验证更新
      console.log('\n🔍 验证更新结果...');
      await new Promise(resolve => setTimeout(resolve, 1000)); // 等待1秒
      
      const verifyResponse = await apiRequest('GET', `/api/v1/users/${createdUserId}`, token);
      if (verifyResponse.success) {
        const updatedUser = verifyResponse.data.data;
        console.log('📊 更新后的用户信息:');
        console.log(`   用户名: ${updatedUser.username}`);
        console.log(`   真实姓名: ${updatedUser.realName}`);
        console.log(`   邮箱: ${updatedUser.email}`);
        console.log(`   手机: ${updatedUser.phoneNumber}`);
        console.log(`   角色: ${updatedUser.role}`);
        console.log(`   备注: ${updatedUser.remark}`);
        
        // 比较更新结果
        console.log('\n📋 更新结果对比:');
        console.log(`   真实姓名: ${updatedUser.realName === updateData.RealName ? '✅' : '❌'} (期望: ${updateData.RealName})`);
        console.log(`   邮箱: ${updatedUser.email === updateData.Email ? '✅' : '❌'} (期望: ${updateData.Email})`);
        console.log(`   手机: ${updatedUser.phoneNumber === updateData.Phone ? '✅' : '❌'} (期望: ${updateData.Phone})`);
        console.log(`   备注: ${updatedUser.remark === updateData.Remark ? '✅' : '❌'} (期望: ${updateData.Remark})`);

        if (updatedUser.realName === updateData.RealName && 
            updatedUser.phoneNumber === updateData.Phone &&
            updatedUser.email === updateData.Email) {
          console.log('\n🎉 用户数据更新测试完全成功！');
        } else {
          console.log('\n⚠️ 用户数据更新不完整，API可能存在实现问题');
        }
      }
    } else {
      console.log('❌ 更新请求失败:', updateResponse.error);
    }

    // 6. 清理测试数据
    console.log('\n🧹 清理测试数据...');
    const deleteResponse = await apiRequest('DELETE', `/api/v1/users/${createdUserId}`, token);
    if (deleteResponse.success) {
      console.log('✅ 测试用户已删除');
    } else {
      console.log('⚠️ 测试用户删除失败，请手动清理:', createdUserId);
    }

  } catch (error) {
    console.error('❌ 测试过程中发生错误:', error.message);
  }
}

// 获取认证Token
async function getAuthToken() {
  return new Promise((resolve, reject) => {
    const data = JSON.stringify({
      username: 'sysadmin',
      password: 'Admin@123456',
      rememberMe: false
    });

    const options = {
      hostname: 'localhost',
      port: 7001,
      path: '/api/v1/auth/login',
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Content-Length': Buffer.byteLength(data)
      }
    };

    const req = https.request(options, (res) => {
      let body = '';
      res.on('data', (chunk) => body += chunk);
      res.on('end', () => {
        try {
          const result = JSON.parse(body);
          if (result.success && result.data && result.data.token) {
            resolve(result.data.token);
          } else {
            reject(new Error('登录失败: ' + body));
          }
        } catch (e) {
          reject(new Error('解析登录响应失败: ' + e.message));
        }
      });
    });

    req.on('error', reject);
    req.write(data);
    req.end();
  });
}

// 通用API请求函数
async function apiRequest(method, path, token, data = null) {
  return new Promise((resolve) => {
    const postData = data ? JSON.stringify(data) : null;
    const options = {
      hostname: 'localhost',
      port: 7001,
      path: path,
      method: method,
      headers: {
        'Authorization': 'Bearer ' + token,
        ...(postData && { 'Content-Type': 'application/json', 'Content-Length': Buffer.byteLength(postData) })
      }
    };

    const req = https.request(options, (res) => {
      let body = '';
      res.on('data', (chunk) => body += chunk);
      res.on('end', () => {
        try {
          const result = res.statusCode >= 200 && res.statusCode < 300;
          resolve({
            success: result,
            statusCode: res.statusCode,
            data: body ? JSON.parse(body) : null,
            error: result ? null : `HTTP ${res.statusCode}: ${body}`
          });
        } catch (e) {
          resolve({
            success: res.statusCode >= 200 && res.statusCode < 300,
            statusCode: res.statusCode,
            data: body,
            error: body || `HTTP ${res.statusCode}`
          });
        }
      });
    });

    req.on('error', (err) => resolve({
      success: false,
      error: err.message
    }));

    if (postData) req.write(postData);
    req.end();
  });
}

simpleUserUpdateTest();